torch/distributed/_tensor/api.py

```diff
@@ -401,7 +401,7 @@ def distribute_tensor(
     device_mesh = get_global_device_mesh() if device_mesh is None else device_mesh
     # convert tensor to the corresponding device type if it's not in that device type
     if not tensor.is_meta:
-        tensor = tensor.to(device_mesh.device_type)
+        tensor = tensor.to(f"{device_mesh.device_type}:{device_mesh.get_rank()}")
```

Comment:
this looks like a DTensor bug?

Reply:
yes, i think so. we should be able to patch this change to main.  i think using just `cuda` instead of `cuda:{...}` is working in the gpu case because device_mesh sets the device

https://github.com/pytorch/pytorch/blob/b96bb2f1a645da0b591e492f5311c53bcd8bf530/torch/distributed/_tensor/device_mesh.py#L151

Reply:
i've removed this change. it's not needed if we override the _shard* or _replicate* functions.

also, passing the rank here breaks the cpu backend: https://github.com/pytorch/pytorch/blob/563d8058f44d5a6324893b4187b91c4cfb134aa7/c10/core/Device.h#LL180C5-L180C37

maybe, the cpu  backend handles the device id through some mechanism other than the device string.